File: mvn-advent-of-code/src/test/java/advent/twenty_fifteen/Day17NoSuchThingAsTooMuchTest.java
package advent.twenty_fifteen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class Day17NoSuchThingAsTooMuchTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day17NoSuchThingAsTooMuch("/twenty_fifteen/Day17-NoSuchThingAsTooMuch-input.txt", 150);

        assertEquals(1304, dp.getPart1Answer().intValue());
        assertEquals(18, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day17NoSuchThingAsTooMuch("/twenty_fifteen/Day17-NoSuchThingAsTooMuch-example#1.txt", 25);

        assertEquals(4, dp.getPart1Answer().intValue());
        assertEquals(3, dp.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_fifteen/Day17NoSuchThingAsTooMuchTest.scala
package advent.twenty_fifteen

import org.scalatest.{FlatSpec, Matchers}

class Day17NoSuchThingAsTooMuchTest extends FlatSpec with Matchers {
  "2015 Day 17 - Input" should "calculate answers" in {
    val dp = new Day17NoSuchThingAsTooMuch("/twenty_fifteen/Day17-NoSuchThingAsTooMuch-input.txt", 150)

    dp.part1Answer should be(1304)
    dp.part2Answer should be(18)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day17NoSuchThingAsTooMuch("/twenty_fifteen/Day17-NoSuchThingAsTooMuch-example#1.txt", 25)

    dp.part1Answer should be(4)
    dp.part2Answer should be(3)
  }
}
